### PR TITLE
[Housekeeping] Port latest fixes from Xamarin.Forms to .NET MAUI

### LINKED ISSUE #10128: Shell.TitleView does not render on iOS 16
### Description

.NET MAUI on iOS 16 does not render a titleview. Versions 15.4 and below do render a title view

First noticed this in forms but it appears to be happening in MAUI as well
https://github.com/xamarin/Xamarin.Forms/issues/15542

### Steps to Reproduce

1. File -> New Maui App
2. Create a TitleView in any page
3. It won't render on an iOS 16 device.

Screenshot from Android Emulator (WORKING):
image

Screenshot from iPhone 12 iOS 16 (NOT WORKING):
84feca32-009a-4343-b131-a01fcd8fbb0d

### Link to public reproduction project repository

https://github.com/edgiardina/Maui-iOS16-TitleViewBug

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16

### Did you find any workaround?

Have not figured out a workaround.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Port latest fixes from Xamarin.Forms to .NET MAUI.

### Issues Fixed

Fixes #10128

### CODE REVIEW

**Review by jfversluis — CHANGES_REQUESTED** [review]
Looks like there is build errors here!

**Review by rmarinho — CHANGES_REQUESTED** [review]
```
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(18,28): error CS0246: The type or namespace name 'TestShell' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Core\Compatibility.ControlGallery.Core.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(18,28): error CS0246: The type or namespace name 'TestShell' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Core\Compatibility.ControlGallery.Core.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(18,28): error CS0246: The type or namespace name 'TestShell' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\Android.UITests\Compatibility.ControlGallery.Android.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(18,28): error CS0246: The type or namespace name 'TestShell' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\Android.UITests\Compatibility.ControlGallery.Android.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(15,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\Android.UITests\Compatibility.ControlGallery.Android.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(16,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\Android.UITests\Compatibility.ControlGallery.Android.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(15,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\Android.UITests\Compatibility.ControlGallery.Android.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(16,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\Android.UITests\Compatibility.ControlGallery.Android.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(18,28): error CS0246: The type or namespace name 'TestShell' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\iOS.UITests\Compatibility.ControlGallery.iOS.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(18,28): error CS0246: The type or namespace name 'TestShell' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\iOS.UITests\Compatibility.ControlGallery.iOS.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(15,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\iOS.UITests\Compatibility.ControlGallery.iOS.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(16,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\iOS.UITests\Compatibility.ControlGallery.iOS.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(15,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\iOS.UITests\Compatibility.ControlGallery.iOS.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(16,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\iOS.UITests\Compatibility.ControlGallery.iOS.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(18,28): error CS0246: The type or namespace name 'TestShell' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\WinUI.UITests\WinUI.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(18,28): error CS0246: The type or namespace name 'TestShell' could not be found (are you missing a using directive or an assembly reference?) [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\WinUI.UITests\WinUI.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(15,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\WinUI.UITests\WinUI.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(16,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\WinUI.UITests\WinUI.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(15,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\WinUI.UITests\WinUI.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(16,28): error CS0103: The name 'UITestCategories' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\WinUI.UITests\WinUI.UITests.csproj]
    271 Warning(s)
    20 Error(s)
```

**Comment by rmarinho** [comment]
Still failing 

```
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(51,26): error CS0103: The name 'Color' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Core\Compatibility.ControlGallery.Core.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(54,26): error CS0103: The name 'Color' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Core\Compatibility.ControlGallery.Core.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(51,26): error CS0103: The name 'Color' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\Android.UITests\Compatibility.ControlGallery.Android.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(54,26): error CS0103: The name 'Color' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\Android.UITests\Compatibility.ControlGallery.Android.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(51,26): error CS0103: The name 'Color' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\iOS.UITests\Compatibility.ControlGallery.iOS.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(54,26): error CS0103: The name 'Color' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\iOS.UITests\Compatibility.ControlGallery.iOS.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15542.cs(51,26): error CS0103: The name 'Color' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\WinUI.UITests\WinUI.UITests.csproj]
D:\a\_work\1\s\src\Compatibility\ControlGallery\src\Issues.Shared\Issue15565.cs(54,26): error CS0103: The name 'Color' does not exist in the current context [D:\a\_work\1\s\src\Compatibility\ControlGallery\test\WinUI.UITests\WinUI.UITests.csproj]
    271 Warning(s)
    8 Error(s)
```

**Review by rmarinho — APPROVED** [review]

File: src/Compatibility/ControlGallery/src/Issues.Shared/Issue15542.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+namespace Microsoft.Maui.Controls.Compatibility.ControlGallery.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 15542, "[Bug] Shell.TitleView does not render on iOS 16",
+		PlatformAffected.iOS)]
+#if UITEST
+	[NUnit.Framework.Category(UITestCategories.Shell)]
+	[NUnit.Framework.Category(UITestCategories.TitleView)]
+#endif
+	public class Issue15542 : TestShell
```
Comment (PureWeen) [inline src/Compatibility/ControlGallery/src/Issues.Shared/Issue15542.cs:20]: Can we port these tests to our `Controls.DeviceTests` ? 

I have a few examples of doing `TitleView` tests here

https://github.com/dotnet/maui/blob/6dc3d240b434b379b03cdd2c788e26e374c07705/src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.cs#L433

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Compatibility/ControlGallery/src/Issues.Shared/Issue15542.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Compatibility/ControlGallery/src/Issues.Shared/Issue15542.cs:20]: We have a device tests now there validating the same https://github.com/dotnet/maui/blob/main/src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.iOS.cs#L205

From https://github.com/dotnet/maui/pull/12834

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs [hunk 2]
```diff
@@ -116,7 +116,7 @@ protected virtual void OnPagePropertyChanged(object sender, PropertyChangedEvent
 			{
 				SearchHandler = Shell.GetSearchHandler(Page);
 			}
-			else if (e.PropertyName == Shell.TitleViewProperty.PropertyName)
+			else if (e.PropertyName == Shell.TitleViewProperty.PropertyName || e.PropertyName == VisualElement.HeightProperty.PropertyName || e.PropertyName == VisualElement.WidthProperty.PropertyName)
```
Comment (PureWeen) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs:119]: Can we change this to use  `e.IsOneOf` ?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs [hunk 3]
```diff
@@ -116,7 +116,7 @@ protected virtual void OnPagePropertyChanged(object sender, PropertyChangedEvent
 			{
 				SearchHandler = Shell.GetSearchHandler(Page);
 			}
-			else if (e.PropertyName == Shell.TitleViewProperty.PropertyName)
+			else if (e.PropertyName == Shell.TitleViewProperty.PropertyName || e.IsOneOf(VisualElement.HeightProperty, VisualElement.WidthProperty))
```
Comment (mandel-macaque) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellPageRendererTracker.cs:119]:
```suggestion
			else if (e.IsOneOf(Shell.TitleViewProperty, VisualElement.HeightProperty, VisualElement.WidthProperty))
```

**Review by PureWeen — APPROVED** [review]

**Review by jfversluis — APPROVED** [review]

**Pull request merged**